src/details/ast.hpp

```diff
@@ -406,32 +441,70 @@ class TableIndex {
     auto index() const -> Expression;
     auto range() const -> minilua::Range;
     auto raw() const -> ts::Node;
+    auto debug_print() const -> std::string;
 };
+/**
+ * class for field_expression nodes
+ */
+class FieldExpression {
+    using FieldExpTuple = std::tuple<std::shared_ptr<Prefix>, Identifier, minilua::Range>;
+    enum TupleIndex { TABLE, PROPERTY, RANGE };
```

Comment:
Instead of creating a tuple and manually naming the indices you should probably just create a helper struct (this can probably also be nested in the class).